### PR TITLE
feat: filter by me on CRUD list view

### PR DESCRIPTION
**Pull request by lilykuang**
### SUMMARY

Implement  filter with `Me` option for list views:
- Annotation Layers (Owner) 
- Charts (Owner, Created By)
- CSS Templates (Created By)
- Dashboards (Owner, Created By)
- Datasets (Owner) 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

- Annotation Layers
annotation-layers
- Charts
Screen Shot 2020-11-12 at 3 40 16 PM
- CSS Templates
Screen Shot 2020-11-12 at 3 46 06 PM
- Dashboards
Screen Shot 2020-11-12 at 3 42 47 PM
- Datasets
Screen Shot 2020-11-12 at 3 41 07 PM

### TEST PLAN

- add new cypress test filter by me 
- update unit test

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/CRUD/utils.tsx [hunk 1]
```diff
@@ -45,13 +47,14 @@ const createFetchResourceMethod = (method: string) => (
...
-
-    return json?.result?.map(
+    const data = json?.result?.map(
       ({ text: label, value }: { text: string; value: any }) => ({
         label,
         value,
       }),
     );
+
+    return options.concat(data);
```
Comment (nytai) [inline superset-frontend/src/views/CRUD/utils.tsx:57]: I believe some of these results are paginated so each request will end up with multiple `me` entries.

Comment (nytai) [reply superset-frontend/src/views/CRUD/utils.tsx:57]: Actually, nvm. Looks like these results are merged in with the existing ones and get properly deduped. 👍

File: superset-frontend/src/views/CRUD/utils.tsx [hunk 2]
```diff
@@ -33,9 +33,11 @@ const createFetchResourceMethod = (method: string) => (
   resource: string,
   relation: string,
   handleError: (error: Response) => void,
+  userId?: string | number,
 ) => async (filterValue = '', pageIndex?: number, pageSize?: number) => {
   const resourceEndpoint = `/api/v1/${resource}/${method}/${relation}`;
-
+  const options =
+    userId && pageIndex === 0 ? [{ label: 'me', value: userId }] : [];
```
Comment (riahk) [inline superset-frontend/src/views/CRUD/utils.tsx:40]: small nit, but should `me` be capitalized?

**Review by riahk — APPROVED** [review]

**Review by willbarrett — APPROVED** [review]
LGTM after rebase

**Pull request merged**